### ISSUE TITLE
Autohotkey Integration

### ISSUE THREAD

**Issue by bigmutty** [OptiKey/OptiKey#687]
Have you considered somehow integrating [Autohotkey](https://www.autohotkey.com/) with OptiKey? It would be nice to be able to put Autohotkey code directly into the action of a DynamicKey. Right now I could write my macros in an Autohotkey script which is running in the background, and assign the correct hotkey to each Dynamickey. Or I could put each macro in a separate script file and assign those to Dynamickeys, but it takes a second or two to load those script files into memory each time you run them.

You can pass code as text to autohotkey.exe via a command line parameter, but I don't know if that would be any faster than passing in a script file.

**Comment by AdamRoden** [comment]
bigmutty, I use AHK all the time. I thought it would be nice to have all those capabilities in OptiKey, but what do you really gain? In the end, it's just an interface via a "hotkey combo". Is there any reason to not run AHK as a stand-alone program? Are you interested in sharing scripts?

**Comment by bigmutty** [comment]
That's true. The only thing you would gain is that you would only have one file to edit. As for script sharing, I would be happy to! I just startled using Optikey, so I haven't written scripts to use with Optikey yet. Most of the scripts I used before discovering Optikey was to aid in remoting my laptop from my Tobii communication device or to aid in using my wheelchair joystick as a mouse.

**Comment by AdamRoden** [comment]
bigmutty, I'm in the same situation and may have the perfect scripts to share. I use my wheelchair joystick to navigate, combined with a single switch for mouse clicks. My favorite scripts are to time my click to be left, middle, or right depending on how long I hold and whether or not I move the mouse, and to dynamically adjust my mouse speed for enhanced precision and speed. I also have my own special keyboard layout to make the most of being limited to an on-screen keyboard.

**Comment by bigmutty** [comment]
Adam, that would be very helpful! I'm new to GitHub. How can I contact you directly?

**Comment by AdamRoden** [comment]
@JuliusSweetland @bigmutty I've found something that could be awesome. If we add the linked code, <https://github.com/amazing-andrew/AutoHotkey.Interop>, to Optikey and follow the usage examples, it looks like we could easily create a dynamic key for executing AutoHotKey scripts, thereby completely unleashing Optikey and it becoming exponentially more powerful. 
Thoughts?

**Comment by JuliusSweetland** [comment]
Definitely. Can it be done in line with the current optikey support for
plugins, i.e. you define a dynamic key which calls the autohotkey
integration passing a set of arguments which dictate the script and script
arguments, or similar?

**Comment by bigmutty** [comment]
That does look awesome! And fairly simple to integrate too! I went back to work 3 weeks ago, so I haven't had much time for other projects. When I get more time on my hands, I'll definitely take a swing at this.

**Comment by Ross-Hale** [comment]
Wow!

**Comment by alvitawa** [comment]
Do any of you know if it is possible to do the inverse? I.e. trigger optikey actions from an ahk script. Specifically, i want to trigger the zoom click menu using a foot pad.

**Comment by AdamRoden** [comment]
@alvitawa It is not possible to trigger Optikey actions from an ahk script.
